run.py

```diff
@@ -13,51 +15,64 @@
 def print_inputs(
     trainer_name: str,
     task_name: str,
-    output_path: Path,
+    output_path: Path | None,
     device: str,
-    seed: int,
     cfg: BaseConfig,
 ):
     print("Running RL with the following inputs:")
     print(f"trainer_name:\t{trainer_name}")
     print(f"task_name:\t{task_name}")
     print(f"output_path:\t{output_path}")
     print(f"device:  \t{device}")
+    print(f"cfg.seed:    \t{cfg.seed}")
 
     # Report on the configuration
     print("\nConfiguration:")
     print(yaml.dump(asdict(cfg), default_flow_style=False))
 
 
-def default_output_path() -> Path:
-    # derive output path from date and time
+def default_output_path(trainer_name: str, task_name: str, seed: int) -> Path:
     now = datetime.datetime.now()
-    return Path(f"output/{now.strftime('%Y_%m_%d/%H_%M_%S')}")
+    date = now.strftime("%Y_%m_%d")
+    time = now.strftime("%H_%M_%S")
+
+    return Path(f"output/{date}/{time}_{task_name}_{trainer_name}_seed_{seed}")
+
+
+def create_cfg(trainer_name: str, seed: int) -> BaseConfig:
+    cfg = create_default_cfg(trainer_name)
+    cfg.seed = seed
+
+    return cfg
 
 
 def main(
-    trainer_name: str, task_name: str, output_path: Path | None, device: str, seed: int
+    trainer_name: str,
+    task_name: str,
+    cfg: BaseConfig,
+    output_path: Path,
+    device: str,
 ):
-    if output_path is None:
-        output_path = default_output_path()
-        if output_path.exists():
-            raise ValueError(f"Output path {output_path} already exists")
+    """Main function to run experiments.
+
+    Args:
+        trainer_name: Name of the trainer to use.
+        task_name: Name of the task to use.
+        cfg: The configuration to use.
+        output_path: Path to save output to.
+        device: Device to run on.
+    """
+    if output_path.exists():
+        raise ValueError(f"Output path {output_path} already exists")
 
     task = create_task(task_name)
     cfg = create_default_cfg(trainer_name)
```

Comment:
I think this line contains a bug. It always overrides the given cfg argument with the default cfg.